src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -397,159 +468,131 @@ namespace AirLoopHVACDOAS {
-                                                          state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum),
-                                                          OutletNodeErrFlag);
+                        break;
+
+                    case ValidEquipListType::CoilCoolingWater:
+                        thisInletNodeNum = WaterCoils::GetCoilInletNode(state, typeNameUC, CompName, InletNodeErrFlag);
+                        thisOutletNodeNum = WaterCoils::GetCoilOutletNode(state, typeNameUC, CompName, OutletNodeErrFlag);
                         thisDOAS.CWCtrlNodeNum = WaterCoils::GetCoilWaterInletNode(state, "COIL:COOLING:WATER", CompName, errorsFound);
                         if (errorsFound) {
-                            ShowContinueError(state,
-                                              "The control node number is not found in " + CurrentModuleObject + " = " +
-                                                  state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum));
+                            ShowContinueError(state, format("The control node number is not found in {} = {}", CurrentModuleObject, CompName));
                         }
                         PlantUtilities::ScanPlantLoopsForObject(
                             state, CompName, DataPlant::PlantEquipmentType::CoilWaterCooling, thisDOAS.CWPlantLoc, errorsFound, _, _, _, _, _);
                         if (errorsFound) { // is this really needed here, program fatals out later on when errorsFound = true
                             ShowFatalError(state, "GetAirLoopDOASInput: Program terminated for previous conditions.");
                         }
-                    } else if (SELECT_CASE_var == "COIL:HEATING:WATER") {
-                        state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).InletNodeNum(CompNum) =
-                            WaterCoils::GetCoilInletNode(state,
-                                                         SELECT_CASE_var,
-                                                         state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum),
-                                                         InletNodeErrFlag);
-                        state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).OutletNodeNum(CompNum) =
-                            WaterCoils::GetCoilOutletNode(state,
-                                                          SELECT_CASE_var,
-                                                          state.dataAirLoop->OutsideAirSys(thisDOAS.m_OASystemNum).ComponentName(CompNum),
-                                                          OutletNodeErrFlag);
+                        break;
+
+                    case ValidEquipListType::CoilHeatingWater:
+                        thisInletNodeNum = WaterCoils::GetCoilInletNode(state, typeNameUC, CompName, InletNodeErrFlag);
```

Comment:
Let me also say something about these passed error flags. There is no need for a separate undifferentiated error flag (i.e., the flag doesn't say what the error _was_ only that there was one) if there are one or more return values that can be interpreted to mean error. In C (and subsequently C++) the value `-1` or the pointer value `nullptr` are generally returned when something is not found. `thisInletNodeNum = -1` has the same information in it as `thisInletNodeNum = -1` and `InletNodeErrFlag = true`. Just something to think about.